llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -5405,7 +5405,14 @@ bool SelectionDAG::isKnownNeverZero(SDValue Op, unsigned Depth) const {
       if (isKnownNeverZero(Op.getOperand(1), Depth + 1) ||
           isKnownNeverZero(Op.getOperand(0), Depth + 1))
         return true;
-    // TODO: There are a lot more cases we can prove for add.
+    if (KnownBits::computeForAddSub(
```

Comment:
Don't we call `computeKnownBits(Op, Depth).isNonZero();` after the switch in this function? That does the same thing as this in a more general way.